lib/active_record/virtual_attributes/virtual_delegates.rb

```diff
@@ -247,13 +247,21 @@ def self.select_from_alias(to_ref, col, to_model_col_name, src_model_id)
                   to_ref.klass.all
                 end
 
+        src_model   = to_ref.active_record
         to_table    = select_from_alias_table(to_ref.klass, src_model_id.relation)
         to_model_id = to_ref.klass.arel_attribute(to_model_col_name, to_table)
         to_column   = to_ref.klass.arel_attribute(col, to_table)
         arel        = query.except(:select).select(to_column).arel
                            .from(to_table)
                            .where(to_model_id.eq(src_model_id))
 
+        # :type is in the reflection definition (meaning it is polymorphic)
```

Comment:
this is the only change

all other changes are either modifying the test schema (model) for demonstrating this issue

Our code roughly maps to [association_scope](https://github.com/rails/rails/blob/5-2-stable/activerecord/lib/active_record/associations/association_scope.rb#L21)

that method is focused on associations (that is to a particular base record) rather than a relation (that is general template for how records interact)

Now that we have added the `type` part, we have all the part implemented (though our `from().where()` construction is a little lacking)